Global middlewares (#345)
go-fuego/fuego: documentation/docs/guides/middlewares.md

You can use them to add functionalities to your routes, such as logging,
authentication, etc.

Middlewares can be registered at 2 levels:

> - **Route middlewares**: applies only on registered routes, easily scopable to a specific group, or several routes.
> - **Global middlewares**: applies on every request, even non-matching routes (useful for CORS for example).

Reviewer: ```suggestion
- **Route middlewares**: apply only on registered routes, easily scopable to a specific group, or several routes.
- **Global middlewares**: apply on every request, even non-matching routes (useful for CORS for example).
```

Or

```suggestion
- **Route middlewares**: applied only on registered routes, easily scopable to a specific group, or several routes.
- **Global middlewares**: applied on every request, even non-matching routes (useful for CORS for example).
```

Reviewer 2: I think this would be nice actually

```suggestion
- **Route middlewares**: applies only to registered routes, is scoped to a specific group, or several routes.
- **Global middlewares**: applied on every request, even non-matching routes (useful for CORS for example).
```

## Route middlewares

You can add middlewares to a single route.
They are treated as an option to the route handler.

> They will be added in the ServeMux in the order they are declared, when registering the route.

Reviewer: ```suggestion
They will be added in the [`http.ServeMux`](https://pkg.go.dev/net/http#ServeMux) in the order they are declared, when registering the route.
```

> ```go title="main.go" showLineNumbers {13-14}

Reviewer: Wow, TIL !! 🤩 Thanks

package main

import (
	"github.com/go-fuego/fuego"
)

func main() {
	s := fuego.NewServer()

	// Declare the middlewares after the route handler
	fuego.Get(s, "/", myController,
		option.QueryInt("page", "The page number"),
		option.Middleware(middleware1),
		option.Middleware(middleware2, middleware3),
	)

	s.Run()
}
```

### Apply on group or server

> To mimic the well-known `Use` method from `chi` and `Gin`, Fuego provides a `Use` method to add middlewares to a Group or Server. They are treated as an option to the server or group handler and will be applied to all routes.

Reviewer: ```suggestion
To mimic the well-known `Use` method from [`chi`](https://pkg.go.dev/github.com/go-chi/chi/v5#Mux.Use) and [`Gin`](https://pkg.go.dev/github.com/gin-gonic/gin#Engine.Use), Fuego provides a `Use` method to add middlewares to a Group or Server. They are treated as an option to the server or group handler and will be applied to all routes.
```

> But we recommend using the `option.Middleware` method for better readability.

Reviewer: ```suggestion
But, we recommend using the `option.Middleware` method for better readability.
```
